apps/keycloak.py

```diff
@@ -0,0 +1,379 @@
+  krb4_config = /etc/krb.conf
+  krb4_realms = /etc/krb.realms
+  kdc_timesync = 1
+  ccache_type = 4
+  forwardable = true
+  proxiable = true
+
+# The following libdefaults parameters are only for Heimdal Kerberos.
+  v4_instance_resolve = false
+  v4_name_convert = {
+    host = {
+      rcmd = host
+      ftp = ftp
+    }
+    plain = {
+      something = something-else
+    }
+  }
+  fcc-mit-ticketflags = true
+
+[realms]
+  OCF.BERKELEY.EDU = {
+    kdc = kerberos.ocf.berkeley.edu
+    admin_server = kerberos.ocf.berkeley.edu
+  }
+
+[domain_realm]
+  .ocf.berkeley.edu = OCF.BERKELEY.EDU
+  ocf.berkeley.edu = OCF.BERKELEY.EDU
+
+[login]
+  krb4_convert = true
+  krb4_get_tickets = false
+"""
+
+helm_values = {
+    "auth": {"existingSecret": "keycloak", "existingSecretKey": "admin-password"},
+    "production": True,
+    "proxy": "edge",
+    "httpRelativePath": "/",
+    "replicaCount": 2,
+    "ingress": {
+        "enabled": True,
+        "ingressClassName": "contour",
+        "annotations": {
+            "cert-manager.io/cluster-issuer": "letsencrypt",
+            "ingress.kubernetes.io/force-ssl-redirect": "true",
+            "kubernetes.io/tls-acme": "true",
+        },
+        "hostname": "idm.ocf.berkeley.edu",
+        "tls": True,
+    },
+    "metrics": {"enabled": True},
+    "keycloakConfigCli": {
+        "enabled": True,
+        "existingConfigmap": "keycloak-config-cli",
+        "cleanupAfterFinished": {
+            "enabled": True,
+            "seconds": 0,
+        },
+    },
+    "postgresql": {"enabled": False},
+    "externalDatabase": {
+        "host": "ocf-keycloak",
+        "port": 5432,
+        "user": "keycloak",
+        "database": "keycloak",
+        "existingSecret": "keycloak.ocf-keycloak.credentials.postgresql.acid.zalan.do",
+        "existingSecretPasswordKey": "password",
+    },
+    "extraVolumes": [
+        {
+            "name": "krb5-conf",
+            "configMap": {
+                "name": "krb5-conf",
+            },
+        }
+    ],
+    "extraVolumeMounts": [
+        {
+            "name": "krb5-conf",
+            "mountPath": "/etc/krb5.conf",
+            "subPath": "krb5.conf",
+        }
+    ],
+}
+
+
+def objects():
+    yield ConfigMap("krb5-conf", data={"krb5.conf": krb5_conf}).build()
+    yield ConfigMap(
+        "keycloak-config-cli",
+        data={"configuration": json.dumps(keycloak_config_cli)},
+    ).build()
+
+    yield {
+        "apiVersion": "acid.zalan.do/v1",
+        "kind": "postgresql",
+        "metadata": {"name": "ocf-keycloak"},
+        "spec": {
+            "teamId": "ocf",
+            "volume": {
+                "size": "32Gi",
+                "storageClass": "rbd-nvme",
+            },
+            "numberOfInstances": 1,
+            "users": {"keycloak": ["superuser", "createdb"]},
+            "databases": {"keycloak": "keycloak"},
+            "postgresql": {"version": "14"},
```

Comment:
Any reason we're deploying a new thing with psql 14 instead of 15? Does the operator / keycloak not support 15?

Reply:
it didn't, but i upgraded it